Include/cpython/frameobject.h

```diff
@@ -4,10 +4,26 @@
 #  error "this header file must not be included directly"
 #endif
 
+/* Starting in CPython 3.11, CPython separates the frame state between the
+ * full "introspection frames" exposed by the Python and C runtime state
+ * introspection APIs, and internal lighter weight "execution frames", which
+ * are simple C structures owned by either the interpreter eval loop (while
+ * executing ordinary functions), by a generator or coroutine object (for
+ * frames that are able to be suspended), or by their corresponding
+ * introspection frame (if an instrospection API has been invoked and the
+ * introspection frame created).
```

Comment:
@markshannon This new comment needs review to make sure I've accurately captured the way the new frame implementation works.